src/paho/mqtt/client.py

```diff
@@ -3084,46 +3308,48 @@ def _packet_handle(self):
         elif cmd == CONNACK:
             return self._handle_connack()
         elif cmd == SUBACK:
-            return self._handle_suback()
+            self._handle_suback()
+            return MQTTErrorCode.MQTT_ERR_SUCCESS
         elif cmd == UNSUBACK:
             return self._handle_unsuback()
         elif cmd == DISCONNECT and self._protocol == MQTTv5:  # only allowed in MQTT 5.0
-            return self._handle_disconnect()
+            self._handle_disconnect()
+            return MQTTErrorCode.MQTT_ERR_SUCCESS
         else:
             # If we don't recognise the command, return an error straight away.
             self._easy_log(MQTT_LOG_ERR, "Error: Unrecognised command %s", cmd)
-            return MQTT_ERR_PROTOCOL
+            return MQTTErrorCode.MQTT_ERR_PROTOCOL
 
-    def _handle_pingreq(self):
+    def _handle_pingreq(self) -> MQTTErrorCode:
         if self._in_packet['remaining_length'] != 0:
-            return MQTT_ERR_PROTOCOL
+            return MQTTErrorCode.MQTT_ERR_PROTOCOL
 
         self._easy_log(MQTT_LOG_DEBUG, "Received PINGREQ")
         return self._send_pingresp()
 
-    def _handle_pingresp(self):
+    def _handle_pingresp(self) -> MQTTErrorCode:
         if self._in_packet['remaining_length'] != 0:
-            return MQTT_ERR_PROTOCOL
+            return MQTTErrorCode.MQTT_ERR_PROTOCOL
 
         # No longer waiting for a PINGRESP.
         self._ping_t = 0
         self._easy_log(MQTT_LOG_DEBUG, "Received PINGRESP")
-        return MQTT_ERR_SUCCESS
+        return MQTTErrorCode.MQTT_ERR_SUCCESS
 
-    def _handle_connack(self):
+    def _handle_connack(self) -> MQTTErrorCode:
         if self._protocol == MQTTv5:
             if self._in_packet['remaining_length'] < 2:
-                return MQTT_ERR_PROTOCOL
+                return MQTTErrorCode.MQTT_ERR_PROTOCOL
         elif self._in_packet['remaining_length'] != 2:
-            return MQTT_ERR_PROTOCOL
+            return MQTTErrorCode.MQTT_ERR_PROTOCOL
 
         if self._protocol == MQTTv5:
             (flags, result) = struct.unpack(
                 "!BB", self._in_packet['packet'][:2])
             if result == 1:
                 # This is probably a failure from a broker that doesn't support
                 # MQTT v5.
-                reason = 132 # Unsupported protocol version
+                reason = ReasonCodes(CONNACK >> 4, aName="Unsupported protocol version")
```

Comment:
This change doesn't seem strictly type-related - would be nice to be sure it's covered by tests, at least?

Reply:
I've done this after adding type to the on_connect callback and saw that reason could be either a int or a ReasonCodes. I believe using int was a mistake.

Anyway I've added a test, and test show that it should no break compatibility (`reason == 132` work)